llvm/include/llvm/IR/FixedMetadataKinds.def

```diff
@@ -53,3 +53,6 @@ LLVM_FIXED_MD_KIND(MD_DIAssignID, "DIAssignID", 38)
 LLVM_FIXED_MD_KIND(MD_coro_outside_frame, "coro.outside.frame", 39)
 LLVM_FIXED_MD_KIND(MD_mmra, "mmra", 40)
 LLVM_FIXED_MD_KIND(MD_noalias_addrspace, "noalias.addrspace", 41)
+// TODO: this will likelly be placed somewhere else,
+// so we don't mix dxil/hlsl/spirv and clang metadata
+LLVM_FIXED_MD_KIND(MD_dxil_controlflow_hints, "dx.controlflow.hints", 42)
```

Comment:
Since this same hint is used for all HLSL targets not just DXIL, we should name it differently and translate it to a DXIL name late. Something like `llvm.branch` would match the pattern used by loop hints.

@nikic, HLSL has hints for backends to indicate when branches should be flattened or preserved. How do you feel about a generic `llvm.branch` metadata?

Reply:
A note about how we handle these attributes today in our fork:

We do nothing in the IR optimizer to force preserving or flattening branches. If the branches these attributes are attached to survive through optimization we encode the attribute into the DXIL that gets handed to the backend. If not, it gets dropped. We don't expect to change that behavior, although we may implement passes to perform flattening/preserving when we look at lowering to ISA.

Reply:
@nikic @llvm-beanz removed the changes in IR Builder. Since this metadata is just passing through the backend, I am using `hlsl.controlflow.hint` and `dx.controlflow.hints`, the latter to keep compatibility with DXIL.